server/src/gateway/gateway.ts

```diff
@@ -137,4 +140,17 @@ export class Gateway implements OnGatewayConnection, OnGatewayDisconnect {
   ): Promise<ObjectResponseObject> {
     return this.objectActionService.handleDeleteObject(client, data);
   }
+
+  @SubscribeMessage('cursor-move')
+  @UseGuards(BoardPermissionGuard)
+  @MinimumBoardPermission(BoardPermission.VIEWER)
+  handleCursorMove(
+    client: GwSocketWithTarget,
+    data: CursorData,
```

Comment:
validations wont work for the interface - it needs to be a class as declared in cursor.data

Reply:
Okay, now it is class